src/utils/fixPrototype.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Replaces prototype, when extending built-in classes (like Error).
+ * Must be called before calling any methods defined in the subclass.
+ * Otherwise, `this` will still think it's an instance of the parent class, which doesn't have a subclass methods.
+ *
+ * Usually used for `instanceof` to work correctly.
+ */
+export const fixPrototype = <T extends object>(
```

Comment:
<img width="889" alt="Screenshot 2025-04-20 at 10 32 46 PM" src="https://github.com/user-attachments/assets/fc551cb9-659c-4e05-863e-1d6b70496edb" />
Do we need this?

Reply:
It indeed works without setting proto:
<img width="609" alt="image" src="https://github.com/user-attachments/assets/589c96a7-4f34-425c-9b51-ae94c5e8023d" />

Thanks for the hint @IrinaStoetski!
Fixed [here](https://github.com/Code-the-Dream-School/ii-practicum-team-2-back/pull/10/commits/6ab21af0233092db9843a39a73c53009b7fbc8f7).